### PR TITLE
Adding support for vanilla Objects

### PR DESCRIPTION
**Pull request by MatApple**

### CODE REVIEW

File: mixins/StoreMixin.js [hunk 1]
```diff
@@ -18,53 +18,53 @@ var ListenersDescription = {
...
+        );
+      }
+    });
+
+    this.store_listeners = storesListeners;
+    this.store_addListeners();
   },
 
-  componentWillUnmount: function () {
+  store_clearListeners: function () {
```
Comment (kennyt) [inline mixins/StoreMixin.js]: why do we need to create this and not just use `store_removeListeners`?

Comment (MatApple) [inline mixins/StoreMixin.js]: You're right, good catch. Fixed.

Comment (kennyt) [inline mixins/StoreMixin.js]: did you push up the changes? :)

File: mixins/StoreMixin.js [hunk 2]
```diff
@@ -18,53 +18,53 @@ var ListenersDescription = {
...
-            events.forEach(function (event) {
-              listener.events[event] =
-                ListenersDescription[storeName].events[event];
-            });
-          }
-
-          storesListeners[storeName] = Util.extend(
-            {}, ListenersDescription[storeName], listener
-          );
+  store_setListeners: function (storeListeners) {
```
Comment (kennyt) [inline mixins/StoreMixin.js]: what do you think about calling this `store_initializeListeners`? `setListeners` to me sounds a lot of like `addListeners` whereas if we had the word initialize in there, it seems like something that should be called once per use

**Comment by kennyt** [comment]
after my one naming comment, I think this looks good

**Pull request merged**